### PR TITLE
Proposed a security.yaml file for the "test" environment

### PR DESCRIPTION
**Pull request by javiereguiluz**
| Q             | A
| ------------- | ---
| License       | MIT

Just a proposal. This is a common practice in Symfony apps and it's was surprising not having it when upgrading the Symfony Demo to use Flex.

### CODE REVIEW

**Comment by Pierstoval** [comment]
Isn't this behavior prohibited by [symfony/security-bundle's configuration](https://github.com/symfony/security-bundle/blob/master/DependencyInjection/MainConfiguration.php#L229) ?

**Comment by javiereguiluz** [comment]
@Pierstoval are you sure? We document this feature: https://symfony.com/doc/current/testing/http_authentication.html and I've been using it for a long time in lots of projects.

**Comment by Pierstoval** [comment]
Oh yeah, nevermind, it's all about new keys, not existing keys, it's ok 😛

**Comment by stof** [comment]
yeah, SecurityBundle let's you reconfigure existing firewalls. What it forbids is adding new firewalls, because merging configuration would make it harder to control the order of firewalls in this case (reconfiguring an existing firewall does not move it in the list)

**Comment by fabpot** [comment]
I don't think this is a good idea. What if there is no `main` firewall or if it is named differently?

**Comment by javiereguiluz** [comment]
The main security.yaml recipe creates a "main" firewall (https://github.com/symfony/recipes/blob/master/symfony/security-bundle/3.3/config/packages/security.yaml#L9). I supposed we could use the same name for the "test" env. Besides, in the Symfony Docs we use the same "main" name everywhere, and my guess is that most people never changes this (only advanced developers ... but they will quickly realize that it must be changed in test/security.yaml too).

**Comment by fabpot** [comment]
Can we at least add a comment like done in the docs?

**Comment by fabpot** [comment]
I would also add a comment in the main security.yml to warn about changing the dev file when renaming the "main" firewall.

**Comment by fabpot** [comment]
@javiereguiluz How do we move forward on this one?

**Comment by javiereguiluz** [comment]
I've added the help comments you asked for. Before merging/rejecting, I'd ask our security experts: @weaverryan and @chalasr. Thanks!

File: symfony/security-bundle/3.3/config/packages/security.yaml [hunk 1]
```diff
@@ -6,6 +6,7 @@ security:
         dev:
             pattern: ^/(_(profiler|wdt)|css|images|js)/
             security: false
+        # if the name of the firewall changes, update it in the packages/test/security.yaml file too
```
Comment (fabpot) [inline symfony/security-bundle/3.3/config/packages/security.yaml]: `if you change the name of the firewall, update it ...`

**Comment by weaverryan** [comment]
I'm not sure about this... this simplifies testing... except if you're testing your authentication itself (and then it's impossible). And if you're system uses some more interesting authentication mechanism (e.g. API tokens, or more simply, a Guard form login where you load User data from some SSO API), then it will fail: this only works if your normal authentication mechanism uses the user provider in the traditional way.

I'd like to give a proactive recommendation, but I'm short on time right this moment. I like the thinking though: how can we *truly* simplify / improve the testing experience?

**Comment by fabpot** [comment]
Didn't see @weaverryan comment, reverting now as I agree with him.

**Comment by Pierstoval** [comment]
@weaverryan if you want to test a Guard, you can still unit test it, that's the big advantage of having the whole authentication process at the same place. The problem comes, as you say it, for other built-in systems, like oauth or `form_login`.

**Pull request merged**